src/i2p/I2PManager.cpp

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2014-2024, The Monero Project
+//
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification, are
+// permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of
+//    conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list
+//    of conditions and the following disclaimer in the documentation and/or other
+//    materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be
+//    used to endorse or promote products derived from this software without specific
+//    prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include <cstring>
+#include <QApplication>
+#include <QDir>
+#include "I2PManager.h"
+
+QString buildI2pdTunnelsConf(bool allowIncomingConnections) {
+    if (!allowIncomingConnections) return QString("");
+    return QString(
+        "[monero-node]\n"
+        "type = server\n"
+        "host = 127.0.0.1\n"
+        "# Anonymous inbound port\n"
+        "port = 1885\n"
+        "inport = 0\n"
+        "keys = monero-mainnet.dat\n"
+        "\n"
+        "[monero-rpc]\n"
+        "type = server\n"
+        "host = 127.0.0.1\n"
+        "# Restricted RPC port\n"
+        "port = 18089\n"
+        "keys = monero-mainnet.dat\n"
```

Comment:
```suggestion
        "keys = monero-mainnet-rpc.dat\n"
```

May want to use a separate key for rpc (more private)